src/middleware/queries.ts

```diff
@@ -47,21 +85,109 @@ export function proposalVotes(
       }
     }
   `;
+  const queryV2 = gql`
+    query ProposalVotesQuery(
+      $pipId: String!
+      $vote: Boolean!
+      $count: Int
+      $skip: Int
+      $orderBy: [ProposalVotesOrderBy!]!
+    ) {
+      proposalVotes(
+        filter: { proposalId: { equalTo: $pipId }, vote: { equalTo: $vote } }
+        first: $count
+        offset: $skip
+        orderBy: $orderBy
+      ) {
+        nodes {
+          blockId
+          eventIdx
+          account
+          vote
+          weight
+        }
+      }
+    }
+  `;
 
   return {
-    query,
-    variables,
+    v1: {
+      query,
+      variables,
+    },
+    v2: {
+      request: {
+        query: queryV2,
+        variables: {
+          ...variables,
+          pipId: variables.pipId.toString(),
+          orderBy: [mapProposalVotesOrder(variables.orderBy)],
+        },
+      },
+      mapper: ({ proposalVotes: results }) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const { nodes } = results!;
+        return {
+          proposalVotes: nodes.map(node => {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { blockId, ...rest } = node!;
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            return { __typename: 'ProposalVote', block_id: +blockId, ...rest };
+          }),
+        };
+      },
+    },
   };
 }
 
+/**
+ *  @hidden
+ */
+function mapProposalVotesOrder(
+  orderInput?: Maybe<ProposalVotesOrderByInput>
+): ProposalVotesOrderBy {
+  if (!orderInput) {
+    return ProposalVotesOrderBy.BlockIdAsc;
+  }
+  const { order, field } = orderInput;
+  if (field === ProposalVotesOrderFields.BlockId) {
+    if (order === Order.Asc) {
+      return ProposalVotesOrderBy.BlockIdAsc;
+    } else {
+      return ProposalVotesOrderBy.BlockIdDesc;
+    }
+  } else if (field === ProposalVotesOrderFields.Vote) {
+    if (order === Order.Asc) {
+      return ProposalVotesOrderBy.VoteAsc;
+    } else {
+      return ProposalVotesOrderBy.VoteDesc;
+    }
+  } else if (field === ProposalVotesOrderFields.Weight) {
+    if (order === Order.Asc) {
+      return ProposalVotesOrderBy.WeightAsc;
+    } else {
+      return ProposalVotesOrderBy.WeightDesc;
+    }
+  }
+  // Not smart enough type system
```

Comment:
I would have thought TS would be smart enough if the checks above were exhaustive.